### ISSUE TITLE
Memory leak in Tomcat

### ISSUE THREAD

**Issue by medvedutka** [orientechnologies/orientdb#4537]
Web-application running on Tomcat 7 which use an embeded (plocal) Orient DB (v2.0.12).
When I shutdown Tomcat (or just undeploy the webapp), I got a lot of severe logs like:

```
GRAVE: The web application [/AP] created a ThreadLocal with key of type [com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.InitOModifiableInteger] (value [com.orientechnologies.common.concur.lock.OReadersWriterSpinLock$InitOModifiableInteger@12c4ef80]) and a value of type [com.orientechnologies.common.types.OModifiableInteger] (value [0]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
juil. 07, 2015 6:01:17 PM org.apache.catalina.loader.WebappClassLoader checkThreadLocalMapForLeaks
GRAVE: The web application [/AP] created a ThreadLocal with key of type [com.orientechnologies.common.concur.lock.OReadersWriterSpinLock.InitOModifiableInteger] (value [com.orientechnologies.common.concur.lock.OReadersWriterSpinLock$InitOModifiableInteger@260903e7]) and a value of type [com.orientechnologies.common.types.OModifiableInteger] (value [0]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
juil. 07, 2015 6:01:17 PM org.apache.catalina.loader.WebappClassLoader checkThreadLocalMapForLeaks
GRAVE: The web application [/AP] created a ThreadLocal with key of type [com.orientechnologies.common.concur.lock.ODistributedCounter.ThreadHashCode] (value [com.orientechnologies.common.concur.lock.ODistributedCounter$ThreadHashCode@266b2f0b]) and a value of type [java.lang.Integer] (value [1710373065]) but failed to remove it when the web application was stopped. Threads are going to be renewed over time to try and avoid a probable memory leak.
juil. 07, 2015 6:01:17 PM org.apache.catalina.loader.WebappClassLoader checkThreadLocalMapForLeaks
...

```

To shutdown the DB, I use "Orient.instance().shutdown();" in a ServletContextListener.contextDestroyed(..) overrided method.

**Comment by laa** [comment]
Could you send memory dump after call of "Orient.instance().shutdown(); ?

**Comment by medvedutka** [comment]
Which kind of memory dump do you want ? How to create it ?
Note: I sent a small maven project which reproduce the bug here:
https://groups.google.com/forum/#!topic/orient-database/bAOK-bVrd3U

**Comment by laa** [comment]
Yes,
I saw it, that is enough information till now ))

**Comment by laa** [comment]
Fixed in 2.1-SNAPSHOT. Thank you for report )).

**Comment by jdillon** [comment]
This has leaked in `com.sun.mangement` packages into the osgi packaging due to use of mbean proxy, suggested fix which should still allow the feature to work w/o the wrinkle:

https://github.com/orientechnologies/orientdb/pull/4580
